action.yml

```diff
@@ -0,0 +1,48 @@
+name: poutine - GitHub Actions SAST
+description: |
+  BoostSecurity.io’s poutine detects vulnerabilities and misconfigurations in your GitHub Actions workflows.
+  Designed to streamline security analysis, poutine scans your repository’s CI/CD pipelines.
+  It offers insights to secure your software supply chain efficiently.
+inputs:
+  path:
+    description: 'Path to the local repository to analyze.'
+    required: true
+    default: '.'
+runs:
+  using: composite
+  steps:
+    - name: Download and Setup
+      id: setup
+      shell: bash
+      env:
+        PINNED_VERSION: v0.9.3
+        PINNED_ARCH: poutine_Linux_x86_64.tar.gz
+        PINNED_HASH: 938c36de89eea18d69c8d5605a64ef1573e682cfd02b61d4cc2de3b0663a63bc
+      run: |
+        # Create a temp dir for the tool and its results
+        POUTINE_TOOL_DIR=$(mktemp -d)
+        cd "${POUTINE_TOOL_DIR}"
+        echo "POUTINE_TOOL_DIR=${POUTINE_TOOL_DIR}" >> "${GITHUB_ENV}"
+        
+        # Download tool archive and verify checksum
+        curl -sLO "https://github.com/boostsecurityio/poutine/releases/download/${PINNED_VERSION}/${PINNED_ARCH}"
+        echo "${PINNED_HASH}  ${PINNED_ARCH}" | shasum -a 256 --check
+        
+        # Decompress the tool
+        tar -xzf "${PINNED_ARCH}"
+        
+    - name: Analyze the repository
+      id: analyze
+      shell: bash
+      run: |
+        POUTINE_SARIF_FILE="${POUTINE_TOOL_DIR}/results.sarif"
+        echo "POUTINE_SARIF_FILE=${POUTINE_SARIF_FILE}" >> "${GITHUB_ENV}"
+        "${POUTINE_TOOL_DIR}/poutine" -format sarif analyze_local "${TARGET_PATH}" > "${POUTINE_SARIF_FILE}"
+      env:
+        TARGET_PATH: ${{ inputs.path }}
+
+    - name: Upload SARIF Findings
```

Comment:
should be optional to upload findings to code scanning alerts? the action will fail otherwise in a private repo without GHAS.